CMakeLists.txt

```diff
@@ -102,27 +118,94 @@ if(USE_CUDA)
   list(APPEND LINK_LIBRARIES gpuxgboost) 
 endif()
 
+
+# flags and sources for R-package
+if(R_LIB)
+  file(GLOB_RECURSE R_SOURCES
+    R-package/src/*.h
+    R-package/src/*.c
+    R-package/src/*.cc
+  )
+  list(APPEND SOURCES ${R_SOURCES})
+endif()
+
 add_library(objxgboost OBJECT ${SOURCES})
 
-# Executable
-add_executable(runxgboost $<TARGET_OBJECTS:objxgboost> src/cli_main.cc)
-set_target_properties(runxgboost PROPERTIES 
-  OUTPUT_NAME xgboost 
-)
-set_output_directory(runxgboost ${PROJECT_SOURCE_DIR})
-target_link_libraries(runxgboost ${LINK_LIBRARIES})
-
-# Shared library
-add_library(xgboost SHARED $<TARGET_OBJECTS:objxgboost>)
-target_link_libraries(xgboost ${LINK_LIBRARIES})
-set_output_directory(xgboost ${PROJECT_SOURCE_DIR}/lib)
-if(MINGW)
-  # remove the 'lib' prefix to conform to windows convention for shared library names
+
```

Comment:
I would like to keep the cmake file is concise and readable as possible. Can some of this be extracted into the utils file or otherwise shortened?